### PR TITLE
[ASDisplayNode] Ensure that nil can never be returned from -measureWithSizeRange:

### PR DESCRIPTION
**Pull request by appleguy**
This can happen in rare cases when multiple relayouts occur while a transition is being measured.

### CODE REVIEW

File: AsyncDisplayKit/ASDisplayNode.mm [hunk 1]
```diff
@@ -642,7 +642,12 @@ - (ASLayout *)measureWithSizeRange:(ASSizeRange)constrainedSize
 {
   ASDN::MutexLocker l(__instanceLock__);
   if (! [self shouldMeasureWithSizeRange:constrainedSize]) {
-    return _calculatedLayout;
+    ASDisplayNodeAssertNotNil(_calculatedLayout, @"-[ASDisplayNode measureWithSizeRange:] _layout should not be nil! %@", self);
+    if (_calculatedLayout != nil) {
+      return _calculatedLayout;
+    } else {
+      return [ASLayout layoutWithLayoutableObject:self constrainedSizeRange:constrainedSize size:CGSizeZero];
+    }
```
Comment (maicki) [inline AsyncDisplayKit/ASDisplayNode.mm]: Just a nitpick, but what about this style:
```objc
return _calculatedLayout ?: [ASLayout layoutWithLayoutableObject:self constrainedSizeRange:constrainedSize size:CGSizeZero];
```

File: AsyncDisplayKit/Private/ASStackUnpositionedLayout.mm [hunk 2]
```diff
@@ -28,7 +28,13 @@
...
   const CGFloat childCrossMin = alignItems == ASStackLayoutAlignItemsStretch ? crossMin : 0;
   const ASSizeRange childSizeRange = directionSizeRange(style.direction, stackMin, stackMax, childCrossMin, crossMax);
-  return [child measureWithSizeRange:childSizeRange];
+  ASLayout *layout = [child measureWithSizeRange:childSizeRange];
+  ASDisplayNodeCAssertNotNil(layout, @"ASLayout returned from measureWithSizeRange: must not be nil: %@", child);
+  if (layout != nil) {
+    return layout;
+  } else {
+    return [ASLayout layoutWithLayoutableObject:child constrainedSizeRange:childSizeRange size:CGSizeZero];
+  }
```
Comment (maicki) [inline AsyncDisplayKit/Private/ASStackUnpositionedLayout.mm]: Same as above ...

**Comment by maicki** [comment]
Just two nitpicks otherwise looks good to me.

**Comment by appleguy** [comment]
@maicki Thanks!  I'd originally done it that way, not sure why I decided to change.  Anyway, put it back to that way.  I do normally put a space like ? : - but we could start doing the other if you prefer.

**Pull request merged**